base/compiler/methodtable.jl

```diff
@@ -40,45 +40,51 @@ end
 getindex(result::MethodLookupResult, idx::Int) = getindex(result.matches, idx)::MethodMatch
 
 """
-    findall(sig::Type, view::MethodTableView; limit=typemax(Int))
+    findall(sig::Type, view::MethodTableView; limit::Int=typemax(Int)) -> MethodLookupResult or missing
 
 Find all methods in the given method table `view` that are applicable to the
 given signature `sig`. If no applicable methods are found, an empty result is
 returned. If the number of applicable methods exceeded the specified limit,
 `missing` is returned.
 """
 function findall(@nospecialize(sig::Type), table::InternalMethodTable; limit::Int=typemax(Int))
-    _min_val = RefValue{UInt}(typemin(UInt))
-    _max_val = RefValue{UInt}(typemax(UInt))
-    _ambig = RefValue{Int32}(0)
-    ms = _methods_by_ftype(sig, nothing, limit, table.world, false, _min_val, _max_val, _ambig)
-    if ms === false
-        return missing
-    end
-    return MethodLookupResult(ms::Vector{Any}, WorldRange(_min_val[], _max_val[]), _ambig[] != 0)
+    return _findall(sig, nothing, table.world, limit)
 end
 
 function findall(@nospecialize(sig::Type), table::OverlayMethodTable; limit::Int=typemax(Int))
+    result = _findall(sig, table.mt, table.world, limit)
+    result === missing && return missing
+    if !isempty(result)
+        if all(match->match.fully_covers, result)
```

Comment:
It only matters if `result[end].fully_covers`

Reply:
Let me check my understanding.
We can just look at `result[end].fully_covers` since:
- `jl_matching_methods` returns matching methods in order of speciality
  (and so the last match is always the most general case)
- if the last match `fully_covers`, it means this call is assured to be 
  dispatched to the last match when it's not dispatched with the other matches